src/azure-cli-core/azure/cli/core/tests/test_util.py

```diff
@@ -213,6 +213,11 @@ def test_configured_default_setter(self):
 
     @mock.patch('azure.cli.core.__version__', '7.8.9')
     def test_get_az_user_agent(self):
+        os.environ['AZ_INSTALLER'] = 'PIP'
+        actual = get_az_user_agent()
+        self.assertEqual(actual, 'AZURECLI/7.8.9 (PIP)')
+
+        del os.environ['AZ_INSTALLER']
```

Comment:
A cleaner way to do the test: 

```suggestion
        with mock.patch.dict('os.environ', {'AZ_INSTALLER': 'PIP'}):
            actual = get_az_user_agent()
            self.assertEqual(actual, 'AZURECLI/7.8.9 (PIP)')

```